### PR TITLE
forms: Emit events for Submit+Reset on forms.

### PR DESCRIPTION
**Pull request by JeanMeche**
This commit add 2 new events sent over the unified event observable.

Fixes #54599

### CODE REVIEW

File: goldens/public-api/forms/index.md [hunk 1]
```diff
@@ -565,6 +565,13 @@ export interface FormRecord<TControl> {
     }): void;
 }
 
+// @public
+export class FormResettedEvent extends ControlEvent {
```
Comment (jnizet) [inline goldens/public-api/forms/index.md:569]: FormResetEvent (resetted is not correct English)

**Review by jnizet — CHANGES_REQUESTED** [review]
It would be nice if the Formgroup could have a getter to tell us if it has been submitted, too

File: goldens/public-api/forms/index.md (hunk 1 above)
Comment (JeanMeche) [reply goldens/public-api/forms/index.md]: Of course, thank you.

**Comment by msmallest** [comment]
> It would be nice if the Formgroup could have a getter to tell us if it has been submitted, too

Yeah, having existing getters for the types of events captured the with this unified events API has been really nice. For example:

```ts
export function touchedEvents$<T>(form: AbstractControl<T>) {
  return form.events.pipe(
    filter(
      (event: ControlEvent): event is TouchedEvent =>
        event instanceof TouchedEvent
    )
  );
}

observableValue$ = this.touchedEvents$(form).pipe(startWith(form.touched))
$signalValue = toSignal(this.observableValue$);
```

For `value`/`status`/`pristine`/`touched` events exposed so far. These initial values allow not just observables but also `toSignal`ed things to have a starting value. Having a getter to get the status of the submit at the time of the `startWith` would be great for things like this, as well as other use cases beyond this.

**Review by dylhunn — APPROVED** [review]
reviewed-for: fw-core, public-api, fw-forms

A nice addition to the new API! I think we need to get this in v18, even though it's technically after feature freeze, since it will be more breaking to land if we wait for v18.1.

**Review by atscott — APPROVED** [review]
reviewed-for: public-api

**Comment by dylhunn** [comment]
Presubmit looks good: https://fusion2.corp.google.com/presubmit/631188647/OCL:631188647:BASE:631188670:1715031403246:38e97a8c/targets

**Pull request closed without merging**